ha/alert/alert_monitor.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+
+import os
+import sys
+import syslog
+import traceback
+
+from ha import const
+from cortx.utils.log import Log
+from cortx.utils.conf_store.conf_store import Conf
+from ha.core.config.config_manager import ConfigManager
+from ha.alert.alert_factory import AlertFactory
+
+# Can be moved to const.py
+REQUIRED_EVENTS = ["node" , "resource"]
+
+class AlertMonitor:
+
+    def __init__(self):
+        """
+        Init alert monitor
+        """
+        super(AlertMonitor, self).__init__()
+        ConfigManager.init("alert_monitor")
+
+        # get environment variables
+        self.crm_env = self.get_env()
+
+    def get_env(self):
+        """
+        Get env variable and parameter provided by pacemaker
+        """
+        try:
+            key = None
+            crm_env = {}
+            env = os.environ
+            for key in env.keys():
+                if key.startswith("CRM_"):
+                    crm_env[key] = env[key]
+            return crm_env
+        except Exception as e:
+            Log.error(e)
+            return crm_env
+
+    def process_alert(self):
+        """
+        Process alert
+        """
+
+        self.validate_pacemaker()
+        self.load_recipient()
+
+        # validate event type and get class
+        try:
+             event_type = self.validate_event()
```

Comment:
I hope the not supported event_type is handled properly.

Reply:
the try catch will catch the same
However , I will add an additional if check there .

Reply:
Logging added